airport-challenge-solution.md

```diff
@@ -0,0 +1,158 @@
+
+**Test 5**: set new Landed Capacity
+**Test 6**: when landedPlanes length equals landedCapacity, do not allow landPlane
+
+## Part 4 User Story
+
+As an air traffic controller
+So I can get passengers on the way to their destination
+I want to instruct the airport to let a plane take off and confirm that it is no longer in the airport
+
+# Part 4 Domain Model
+
+| Objects | Properties                  | Messages                    | Outputs  |
+| ------- | --------------------------- | --------------------------- | -------- |
+| Airport | landedPlanes @Array[@Plane] | landPlane(@Plane)           | @Void    |
+|         | landedCapacity @Integer     | isAirportFull()             | @Boolean |
+|         |                             | getLandedCapacity()         | @Integer |
+|         |                             | newLandedCapacity(@Integer) | @Void    |
+|         |                             | getLandedPlaneCount()       | @Integer |
+|         |                             | takeOffPlane(@Plane)        | @Void    |
+|         |                             | isPlanePresent(@Plane)      | @Boolean |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests - 7&8
+
+7 landPlane Plane(id) and check if isPlanePresent (T/F)
+8 in the same way - takeOffPlane Plane(id) and check if isPlanePresent (T/F)
+
+## Part 5 User Story
+
+As an air traffic controller
+To avoid confusion
+I want to prevent asking the airport to let planes take-off which are not at the airport, or land a plane that's already landed
+
+## Part 5 Domain Model (not added to) - already fine from above
+
+| Objects | Properties                  | Messages                    | Outputs  |
+| ------- | --------------------------- | --------------------------- | -------- | ----------- |
+| Airport | landedPlanes @Array[@Plane] | landPlane(@Plane)           | @Void    |
+|         | landedCapacity @Integer     | isAirportFull()             | @Boolean | did not use |
+|         |                             | getLandedCapacity()         | @Integer |
+|         |                             | newLandedCapacity(@Integer) | @Void    |
+|         |                             | getLandedPlaneCount()       | @Integer |
+|         |                             | takeOffPlane(@Plane)        | @Void    |
+|         |                             | isPlanePresent(@Plane)      | @Boolean |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests 9&10
+
+9. try to take off a plane(id) which is not in the airport (landedPlanes)
+10. try land a second instance of a plane(id) that is already in landedPlanes
+    (need to add a Boolean return to landPlane function?, yes but did not end up using)
+
+## Testing Edge Cases
+
+Have already shown it is not possible to land something that is not a plane - a string and a value, in test 3.
+Have also shown that a plane that is already landed can not land a second time, in test 10.
+
```

Comment:
Read ahead to look at explanations for Edge cases, to understand if I am attempting them as part of the basic criteria. Also need to understand Edge cases more anyway.